### PR TITLE
Harmony 1218

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1218

## Description
Adds an endpoint to generate bucket policies that will allow Harmony to write output into the given bucket/folder.

## Local Test Steps
Will return message saying it only works for AWS deployments, so it has to be tested in the sandbox.
1. deploy this branch to the sandbox
2. set up a bucket in SIT for testing
3. Set up an environment in Insomnia or similar to query your sandbox harmony
4. query the endpoint to generate a bucket policy (change my-staging-bucket to your bucket name)
{{ _.root }}/staging-bucket-policy?bucketPath=my-staging-bucket
5. Edit the bucket policy on your staging bucket from step 2 and replace it with the generated one
6. Issue the following query specifying the staging bucket for your output
{{ _.root }}/C1234208438-POCLOUD/ogc-api-coverages/1.0.0/collections/bathymetry/coverage/rangeset?concatenate=true&subset=lon(-160%3A160)&subset=lat(-80%3A80)&skipPreview=true&maxResults=3&destinationUrl=s3%3A%2F%2Fmy-staging-bucket
7. Verify that their is a folder in your bucket named after the request ID, and that it has a file called `harmony-jog-status-link` in it as well as another folder named after a work-item ID. The final data file should be in that folder.

OPTIONAL: repeat steps 4-7 using a sub-folder, e.g.,`my-bucket/folder-name` and again using a full S3 url instead of jus `bucket-name/folder-name`, e.g., `s3://my-bucket/folder-name`.

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

**Review by ygliuvt — APPROVED** [review]
Two comments (comments only, no need to take action for this PR):
1. It would be good to have an easily accessible documentation page for non-OGC routes, where we can document this and other non-OGC routes. I can see users get tripped by having to encode `/` in the provided bucket path value for this route.
2. NIT: for completeness, we could add a test for `s3://bucket-name/path` as the bucket path value which is also supported.

**Comment by indiejames** [comment]
> Two comments (comments only, no need to take action for this PR):
> 
> 1. It would be good to have an easily accessible documentation page for non-OGC routes, where we can document this and other non-OGC routes. I can see users get tripped by having to encode `/` in the provided bucket path value for this route.
> 2. NIT: for completeness, we could add a test for `s3://bucket-name/path` as the bucket path value which is also supported.

Additional test added

**Review by chris-durbin — APPROVED** [review]
Looks good - tested everything out in the sandbox. The validations look good. I could use an S3 url with a path, just a bucket name, or a bucket name and path. I applied the policies to a test bucket and verified that only the specified path in the bucket could be successfully used for external S3 delivery. Other prefixes in the same bucket received errors writing to them as desired.

File: fixtures/sts.amazonaws.com-443/167658623479245801 [hunk 1]
```diff
@@ -0,0 +1,22 @@
+POST /
```
[the remark this answers is not in the document]
Comment (indiejames) [reply fixtures/sts.amazonaws.com-443/167658623479245801]: I tried to clean these up by removing them and running `npm run test`, but it still created four fixtures.

**Pull request merged**